cgroup.c

```diff
@@ -123,19 +124,43 @@ static int rm_cgroup(const char *fpath, const struct stat *sb, int tflag, struct
 /* If bst has entered a cgroup this function will epoll the cgroup.events file
    to detect when all pids have exited the cgroup ("populated 0"). The cgroup is
    destroyed when this condition is met. */
-static void run_cleaner_child(int cgroupfd, int parentfd, const char *name)
+static void run_cleaner_child(int lock, int parentfd, const char *name)
 {
+	/* Wait for the parent to die before proceeding */
+	int ok;
+	switch (read(lock, &ok, sizeof (ok))) {
+	case -1:
+		warn("run_cgroup_child: read on lock");
+		goto lastDitchEffort;
```

Comment:
Is there really a preference for `goto lastDistchEffort` over `return fallback_delete_cgroup` ? I suppose the former keeps the scope quite a bit tighter since `lastDitschEffort` is only accessable from inside the function, but it feels anachronistic

Reply:
Yeah, it usually ends up being cleaner because the alternative is exfiltrating a bunch of local variables back to the parent, though in this case for one of the two goto labels, the caller has the parentfd.

It's probably one of the two legitimate uses of goto in C code. Thankfully, we have defer in Go :stuck_out_tongue: